### PR TITLE
Update speechbrain to use develop branch

### PR DESCRIPTION
**Pull request by osanseviero**
cc @TParcollet

### CODE REVIEW

**Comment by TParcollet** [comment]
The question I have here is the following: at some point, we discussed switching to defining the interface to use in the config.json. Is it implemented? Cuz here, I see that we instantiate the corresponding interfaces directly.

**Comment by TParcollet** [comment]
Just trying to figure out if https://huggingface.co/speechbrain/emotion-recognition-wav2vec2-IEMOCAP doesn't work just because we currently are building the docker from the Master branch or if we will have to do something on the API side later as well.

File: api-inference-community/docker_images/speechbrain/app/pipelines/automatic_speech_recognition.py [hunk 1]
```diff
@@ -4,20 +4,19 @@
...
         model_type = get_type(model_id)
         if model_type is ModelType.ENCODERASR:
-            # self.model = EncoderASR.from_hparams(source=model_id)
-            raise ValueError("EncoderASR is not a supported interface at the moment")
+            self.model = EncoderASR.from_hparams(source=model_id)
         elif model_type is ModelType.ENCODERDECODERASR:
             self.model = EncoderDecoderASR.from_hparams(source=model_id)
 
             # Reduce latency
-            self.model.modules.decoder.beam_size = 1
```
[the remark this answers is not in the document]
Comment (osanseviero) [reply api-inference-community/docker_images/speechbrain/app/pipelines/automatic_speech_recognition.py:20]: Hmm in this PR the line was changed to `self.model.mods.decoder.beam_size = 1`

**Comment by osanseviero** [comment]
@TParcollet we still need to check that the interface is valid and we also need to know the interface to use the corresponding method.For example, in `audio-to-audio` we need to call either `separate_batch` or `enhance_batch`, etc

**Comment by osanseviero** [comment]
What this will entail is that if you add new interfaces you will still need to do a super small PR here to enable the new interface.

**Pull request merged**